vmss-prototype/vmss-prototype

```diff
@@ -736,20 +736,6 @@ def vmss_prototype_update(sub_args):
                     '--no-wait'
                     ], retries=3, check=False, retry_func=not_found_no_retry)
 
-            finally:
```

Comment:
This is wrong.  We still need this finally block for the case where we fail during the drain.

We should maybe set a flag that the finally block triggers on to not uncordon/annotate the node once it was deleted.

There are exit paths in multiple places where this is needed before the delete-instances is called and succeeds.

Reply:
Got it. Fixed this up so that we track when we've successfully deleted the node, and if we do, we don't initiate the uncordon and annotation hygiene.